src/routes/ui/templates.rs

```diff
@@ -5,6 +5,89 @@ use rocket::request::FlashMessage;
 use routes::ui::view_helpers::*;
 use std::ops::Deref;
 
+macro_rules! HtmlTemplate {
+    ($x:tt) => {{
+        $x {
+            _parent: BaseTemplate {
+                revision: ::GIT_REV,
+            }
+        }
+    }};
+
+    ($x:tt, $flash: ident) => {{
+        $x {
+            _parent: BaseTemplate {
+                flash: $flash,
+                revision: ::GIT_REV,
+            }
+        }
+    }};
+
+    ($x:tt, { $( $y:ident : $z:expr ),* }) => {{
+        $x {
+            $( $y: $z ),*
+            ,_parent: BaseTemplate {
+                flash: None,
+                revision: ::GIT_REV,
+            }
+        }
+    }};
+    ($x:tt, $flash: ident, { $( $y:ident : $z:expr ),* }) => {{
+        $x {
+            $( $y: $z ),*
+            ,_parent: BaseTemplate {
+                flash: $flash,
+                revision: ::GIT_REV,
+            }
+        }
+    }};
+}
+
+macro_rules! PerhapsHtmlTemplate {
```

Comment:
I would almost prefer to wrap `HtmlTemplate` with `Ok(Some(_))` than have a separate macro.

Note, though, if we do end up having a separate macro, you don't need to duplicate all this code -- macros can call other macros, so you can just call to `HtmlTemplate` inside `PerhapsHtmlTemplate`.

In fact, you should be able to just do
```rust
macro_rules! PerhapsHtmlTemplate {
    $($x:tt)* => {{ Ok(Some(HtmlTemplate!($($x)*))) }}
}
```

Reply:
Perfect, thanks - I saw something that implied that macros calling other macros required some other lazy macro crate, which caused me to stop looking. Shoulda just tried it.

Reply:
(Done)